src/util.ts

```diff
@@ -0,0 +1,18 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+export interface PartialList<T> extends Array<T> {
+    nextLink?: string;
+}
+
+export async function listAll<T>(client: { listNext(nextPageLink: string): Promise<PartialList<T>>; }, first: Promise<PartialList<T>>): Promise<T[]> {
```

Comment:
If the user has a ton of items, I don't think they want to wait for them all to load (who knows how long that'll be). Ideally we would have some sort of 'Load more...' functionality, but I don't know if that's worth implementing right now. In the mean time, what if we just showed a warning message saying 'Not all subscriptions are displayed'? That way we don't risk the product looking like it's unresponsive

Reply:
We should request VSC to provide means to show progress indicator so the users can know data is being loaded. Missing data (and no way of accessing them) doesn't seem to be a better option. Other options like pre-load data in the background can also be considered.

Reply:
A progress indicator is nice, but doesn't help the main issue. If I have a ton of resources and we load them all at once, I might not see _any_ resources for several minutes. Partial data is better than no data in this case. I agree it sucks if you have no way to access the rest of the data, but there's always a 'workaround' to use the portal. 

I'm fine with this code as-is if you create an issue to add the "Load more..." functionality later. I think we agree that's the best behavior? The user can see resources immediately and they still have a way to access all resources if they want